test/parallel/test-crypto-engine.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+const common = require('../common');
+
+if (!common.hasCrypto) {
+  common.skip('missing crypto');
+  return;
+}
+
+const assert = require('assert');
+const crypto = require('crypto');
+const fs = require('fs');
+const path = require('path');
+
+let found = 0;
+
+// check at least one builtin engine exists.
+crypto.getEngines().forEach((e) => {
+  if (e.id === 'dynamic')
+    found++;
+});
+
+assert.strictEqual(found, 1);
+
+// check set and get node test engine of
+// test/fixtures/openssl_test_engine/node_test_engine.c
+
+if (process.config.variables.node_shared_openssl) {
+  common.skip('node test engine cannot be built in shared openssl');
+  return;
+}
+
+if (process.config.variables.openssl_fips) {
+  common.skip('node test engine cannot be built in FIPS mode.');
+  return;
+}
+
+let engine_lib;
+
+if (common.isWindows) {
+  engine_lib = 'node_test_engine.dll';
+} else if (common.isAix) {
+  engine_lib = 'libnode_test_engine.a';
+} else if (process.platform === 'darwin') {
+  engine_lib = 'node_test_engine.so';
+} else {
+  engine_lib = 'libnode_test_engine.so';
+}
+
+const test_engine_id = 'node_test_engine';
+const test_engine_name = 'Node Test Engine for OpenSSL';
+const test_engine = path.join(path.dirname(process.execPath), engine_lib);
+
+assert.doesNotThrow(function() {
+  fs.accessSync(test_engine);
+}, 'node test engine ' + test_engine + ' is not found.');
+
+crypto.setEngine(test_engine);
+
+crypto.getEngines().forEach((e) => {
```

Comment:
Is there any reason to use filter rather than forEach? It seems to be equivalent for me.